### PR TITLE
Weather w/ db

### PR DESCRIPTION
**Pull request by BrycePearce**
migrates weather to use sqlite instead of json file

### CODE REVIEW

File: src/commands/weather/command.ts [hunk 1]
```diff
@@ -29,48 +29,52 @@ const command: Command = {
...
-      message.channel.send('Set your default location with .weather set YOUR_LOCATION');
-    }
-
     try {
-      const geoData = await getGeoLocation(userLocation);
-      if (!geoData) {
-        message.channel.send('Location was not found!', { files: [await getRandomEmotePath()] });
+      let user = await DI.userRepository.findOne(message.author.id);
+      const isUpdatingLocation = args[0]?.toLowerCase().trim() === 'set';
+      const location = await getUserLocation(user, args, isUpdatingLocation);
```
Comment (m4tta) [inline src/commands/weather/command.ts:35]: I feel like this `getUserLocation` is doing work it shouldn't be doing inside of it. Like parsing additional args. Perhaps it shouldn't even exist.

You can use the main body of the `execute` method to process subcommands like done here:
https://github.com/BrycePearce/KyuuBot/blob/main/src/commands/reminder/command.ts#L64

File: src/database/entities/User.ts [hunk 2]
```diff
@@ -12,4 +11,7 @@ export class User extends BaseEntity {
 
   @OneToMany(() => Reminder, (reminder) => reminder.user, { cascade: [Cascade.ALL] })
   reminders = new Collection<Reminder>(this);
+
+  @ManyToOne(() => Location, { nullable: true })
```
[the remark this answers is not in the document]
Comment (BrycePearce) [reply src/database/entities/User.ts]: put it on root User entity

File: src/commands/weather/command.ts (hunk 1 above)
Comment (BrycePearce) [reply src/commands/weather/command.ts:40]: Not finding a great way to do this without duplicating code

Comment (BrycePearce) [reply src/commands/weather/command.ts]: refactored it a bit

**Pull request merged**